### PR TITLE
Ensure resolution order is consistent.

### LINKED ISSUE #45: Objects have inconsistent resolution orders
As exposed with zope.interface 5.0.

```
InconsistentResolutionOrderError: For object <implementedBy zope.component.globalregistry.BaseGlobalComponents>.
Base ROs:
{<InterfaceClass zope.interface.interfaces.IComponentLookup>: [<InterfaceClass zope.interface.interfaces.IComponentLookup>,
                                                               <InterfaceClass zope.interface.Interface>],
 <implementedBy zope.interface.registry.Components>: [<implementedBy zope.interface.registry.Components>,
                                                      <InterfaceClass zope.interface.interfaces.IComponents>,
                                                      <InterfaceClass zope.interface.interfaces.IComponentLookup>,
                                                      <InterfaceClass zope.interface.interfaces.IComponentRegistry>,
                                                      <implementedBy builtins.object>,
                                                      <InterfaceClass zope.interface.Interface>]}
Conflict Location:
[[<InterfaceClass zope.interface.interfaces.IComponentLookup>,
  <InterfaceClass zope.interface.Interface>],
 [<implementedBy zope.interface.registry.Components>,
  <InterfaceClass zope.interface.interfaces.IComponents>,
  <InterfaceClass zope.interface.interfaces.IComponentLookup>,
  <InterfaceClass zope.interface.interfaces.IComponentRegistry>,
  <implementedBy builtins.object>,
  <InterfaceClass zope.interface.Interface>],
 [<InterfaceClass zope.interface.interfaces.IComponentLookup>,
  <implementedBy zope.interface.registry.Components>]]
```

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #45.

Tests won't be green until zope.configuration is released with its corresponding fix, as well as zope.security (which needs BTrees).

### CODE REVIEW

File: src/zope/component/tests/test_globalregistry.py [hunk 1]
```diff
@@ -244,3 +244,31 @@ class IFoo(Interface):
         self.assertEqual(list(hr.required), [IFoo])
         self.assertEqual(hr.name, '')
         self.assertTrue(hr.factory is _handler)
+
+
+class TestBaseGlobalComponents(unittest.TestCase):
+
+    from zope.component.testing import setUp, tearDown
```
Comment (mgedmin) [inline src/zope/component/tests/test_globalregistry.py]: This is delightfully evil.

File: tox.ini [hunk 2]
```diff
@@ -28,24 +30,20 @@ deps =
...
     PURE_PYTHON = 1
     PIP_CACHE_DIR = {envdir}/.cache
 
 [testenv:py27-pure]
 basepython =
     python3.4
 setenv =
+    {[testenv]setenv}
     PURE_PYTHON = 1
     PIP_CACHE_DIR = {envdir}/.cache
```
Comment (mgedmin) [inline tox.ini]: Incidentally, I think we can drop PIP_CACHE_DIR overrides, now that PURE_PYTHON is checked at runtime everywhere?

... assuming it is checked at runtime everywhere; I see BTrees (and zodbpickle, and zope.container, and some Zope-2-only packages like Acquisition/Persistence) still have some kind of PURE_PYTHON checks in their setup.py.  *sigh*.

File: tox.ini [hunk 3]
```diff
@@ -28,24 +30,20 @@ deps =
 commands =
     zope-testrunner --test-path=src
 
-[testenv:py34-pure]
-basepython =
-    python3.4
-setenv =
-    PURE_PYTHON = 1
-    PIP_CACHE_DIR = {envdir}/.cache
```
Comment (mgedmin) [inline tox.ini]: This is the sort of thing that could very nicely be handled using [tox's factors](https://tox.readthedocs.io/en/latest/config.html#factors-and-factor-conditional-setting) without multiple [testenv:...] sections.

```ini
[testenv]
setenv =
    ZOPE_INTERFACE_STRICT_IRO=1
    pure: PURE_PYTHON=1
    pure: PIP_CACHE_DIR={envdir}/.cache
```

Comment (jamadden) [reply tox.ini]: Thanks for the link. I've seen that in one other package but hadn't bothered to find the docs yet.

**Comment by jamadden** [comment]
Thanks for the review.

I got the tests all passing, but I'm afraid CI isn't going to be completely happy until `zope.security` is available because coverage falls without it. And `zope.security` won't be green until `zope.component` is released; it has a hard dependency, starting with `zope.security` -> `zope.location` -> `zope.component`. So I think I have to release `zope.component` with green tests and temporarily decreased coverage.

**Pull request merged**